_episodes/twir/2022-06-06-twir-444.md

```diff
@@ -0,0 +1,202 @@
+      keynote](https://mirror.linux.org.au/pub/linux.conf.au/2013/mp4/Keynote_Andrew_Bunnie_Huang.mp4)
+      discussing [Chumby](https://en.wikipedia.org/wiki/Chumby) and creating a hardware startup
+  - [@20:20] - [betrusted.io](https://betrusted.io/), a secure communications system that
+      runs the [Xous microkernel operating  system](https://betrusted.io/xous-book/)
+  - [@21:07] - Tim: Security-critical applications have issues when they ... rely on Rust.
+      There's one quote I want to pull out of the post, which is: "I’m not sure
+      if there is even a good solution to this problem, but, if you are
+      super-paranoid and your goal is to be able to build trustable firmware, be
+      wary of Rust’s expansive software supply chain attack surface!"
+  - [@26:09] - Sean: bunnie I think that you are absolutely, totally, qualified.
+  - [@30:17] - Allen: I did see a macro that he put in there. ... I forget
+    extact. It was very crazy and I was like, "Come on, no one's every going to
+    write something crazy like this" and then I took a look at the RFC that
+    Sean's gonna do and in the comments there was a crazy one like that and I
+    was like, "oh wow, this guy's point's valid".
+  - [@30:49] - [Hyrum's Law](https://www.hyrumslaw.com/), named after [Hyrum
+    Wright](https://twitter.com/hyrumwright).
+    > With a sufficient number of users of an API, it does not matter what you
+    > promise in the contract: all observable behaviors of your system will be
+    > depended on by somebody.
+
+- [@31:50] [Fixing memory leaks](https://onesignal.com/blog/solving-memory-leaks-in-rust/
+) by [Lily Mara](https://twitter.com/TheLily_Mara)
+  - [@34:01] - [`tracing`](https://docs.rs/tracing) crate, created as part of
+   the [tokio](https://tokio.rs/tokio/topics/tracing) project
+  - [@32:33] - "[Is it possible to cause a memory leak in
+    Rust?](https://stackoverflow.com/q/55553048/395287)" - Stack Overflow
+  - [@33:06] -
+    [`std::ops::Drop`](https://doc.rust-lang.org/std/ops/trait.Drop.html) trait
+    documentation
+  - [`std::mem::forget`](https://doc.rust-lang.org/std/mem/fn.forget.html) and
+    [`Box::leak`](https://doc.rust-lang.org/std/boxed/struct.Box.html#method.leak)
+    for intentionally leaking memory
+  - [Out-of memory (OOM) killer internals page](https://linux-mm.org/OOM_Killer) from the [Linux memory
+    management wiki](https://linux-mm.org/)
+  - [@37:54] [`tracing::instrument::Instrument` trait](https://docs.rs/tracing/0.1.34/tracing/instrument/trait.Instrument.html), which fixes this issue
+
+- [@41:29] [Building a Cloud Database from Scratch: Why We Moved from C++ to
+  Rust](https://singularity-data.com/blog/building-a-cloud-database-from-scratch-why-we-moved-from-cpp-to-rust/)
+  by [Yingjun Wu](https://twitter.com/YingjunWu)
+  - [GAT (generic associated traits)](https://blog.rust-lang.org/2021/08/03/GATs-stabilization-push.html)
+  - Allen: [C++ vs Rust] is like apples vs apple pie.
```

Comment:
Was this intended to be a link?

Reply:
Ah yes, but I've forgotten what I meant to refer to..

Reply:
Actually, no it's not. The square brackets are there to provide enough context for the reader.